cosmwasm/register-its-token.ts

```diff
@@ -0,0 +1,252 @@
+    const provider = getDefaultProvider(rpc);
+    const token = new Contract(tokenAddress, IInterchainToken.abi, provider);
+    return await token.totalSupply();
+}
+
+async function registerToken(
+    interchainTokenServiceAddress: string,
+    client: ClientManager,
+    tokenDataToRegister: TokenDataToRegister,
+    dryRun: boolean,
+) {
+    const supply = tokenDataToRegister.supply;
+    const supplyParam = supply ? { tracked: String(supply) } : 'untracked';
+    const msg = {
+        register_p2p_token_instance: {
+            chain: tokenDataToRegister.axelarId,
+            token_id: tokenDataToRegister.tokenId.slice(2),
+            origin_chain: tokenDataToRegister.originChain,
+            decimals: tokenDataToRegister.decimals,
+            supply: supplyParam,
+        },
+    };
+
+    const [account] = await client.accounts;
+    printInfo('Registering token ', JSON.stringify(msg.register_p2p_token_instance));
+
+    // TODO tkulik: uncomment to implement the registration
+    // if (!dryRun) {
+    //     await client.execute(account.address, interchainTokenServiceAddress, msg, 'auto');
+    // }
+}
+
+async function forEachToken(
+    config: ConfigManager,
+    options,
+    processToken: (tokenData: SquidToken, tokenOnChain: SquidTokenData) => Promise<void>,
+) {
+    const { env, tokenIds, chains, squid } = options;
+    const tokenIdsToProcess = new Set(tokenIds);
+    const chainsToProcess = new Set(chains);
+    const tokenInfoString = fs.readFileSync(
+        `axelar-chains-config/info/tokens-p2p/${squid ? 'squid-tokens' : 'tokens'}-${env}.json`,
+        'utf8',
+    );
+    const tokenInfo = JSON.parse(tokenInfoString) as SquidTokenInfoFile;
+    const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+    if (!interchainTokenServiceAddress) {
+        throw new Error('InterchainTokenService contract address not found');
+    }
+    const promises = Object.values(tokenInfo.tokens)
+        .filter((tokenData: SquidToken) => (tokenIds ? tokenIdsToProcess.has(tokenData.tokenId) : true))
+        .flatMap((tokenData: SquidToken) => {
+            return tokenData.chains
+                .filter((chain: SquidTokenData) => {
+                    try {
+                        return (
+                            tokenData.tokenType === 'interchain' &&
+                            (chains ? chainsToProcess.has(chain.axelarChainId) : true) &&
+                            (chain.track ?? true) &&
+                            chain.axelarChainId !== tokenData.originAxelarChainId &&
+                            (chain.registered ? !chain.registered : true) &&
+                            isConsensusChain(config.getChainConfigByAxelarIdLowercase(chain.axelarChainId))
+                        );
+                    } catch (e) {
+                        printError(`Error getting chain config for ${chain.axelarChainId} (skipping chain): ${e.message}`);
```

Comment:
this is a weird error to check/raise at this stage. we should expect that chains are already valid, and the script that created the config did it's job well